From e1c811c3926b43dfdad64f677d73c890a0727a70 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Fri, 20 Sep 2024 17:48:15 +0200
Subject: [PATCH 693/841] feat(stm32mp2): improve BL31 size management
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Change the size of BL31 limit allocation to be half the sysram size.
Defining BL31_PROGBITS_LIMIT to detect overflows.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 6c0c6fd27f43d6541f8597c484b4ce3cff15eb2c feat(stm32mp2): improve BL31 size management]
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Change-Id: Iccc1cc0826b8113a3c2fd6ffa77ca419795854d3
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/406039
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
---
 plat/st/stm32mp2/include/platform_def.h | 2 +-
 plat/st/stm32mp2/stm32mp2_def.h         | 8 ++++++--
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/plat/st/stm32mp2/include/platform_def.h b/plat/st/stm32mp2/include/platform_def.h
index fb2315590..20b81410a 100644
--- a/plat/st/stm32mp2/include/platform_def.h
+++ b/plat/st/stm32mp2/include/platform_def.h
@@ -74,7 +74,7 @@
  * BL31 specific defines.
  ******************************************************************************/
 #define BL31_BASE			0
-#define BL31_LIMIT			STM32MP_BL31_SIZE
+#define BL31_LIMIT			(STM32MP_SEC_SYSRAM_SIZE / 2)
 
 /*******************************************************************************
  * BL33 specific defines.
diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index ae6181a67..288362a69 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -215,8 +215,6 @@ enum ddr_type {
 						 (STM32MP_PARAM_LOAD_SIZE +	\
 						  STM32MP_HEADER_SIZE))
 
-#define STM32MP_BL31_SIZE		U(0x0001F000)	/* 124 KB for BL31 */
-
 #if TRUSTED_BOARD_BOOT
 #if STM32MP_UART_PROGRAMMER || STM32MP_USB_PROGRAMMER
 #define STM32MP_BL2_RO_SIZE		U(0x00021000)	/* 132 KB */
@@ -235,6 +233,12 @@ enum ddr_type {
 #endif /* STM32MP_UART_PROGRAMMER || STM32MP_USB_PROGRAMMER */
 #endif
 
+/* Allocate remaining sysram to BL31 Binary only */
+#define STM32MP_BL31_SIZE		(STM32MP_SEC_SYSRAM_SIZE - \
+					 STM32MP_BL2_SIZE)
+
+#define BL31_PROGBITS_LIMIT		STM32MP_BL31_SIZE
+
 #define STM32MP_BL2_BASE		(STM32MP_SEC_SYSRAM_BASE + \
 					 STM32MP_SEC_SYSRAM_SIZE - \
 					 STM32MP_BL2_SIZE)
-- 
2.47.2

